Metalama.Framework.Engine/Templating/TemplateSyntaxFactoryImpl.cs

```diff
@@ -336,6 +336,45 @@ public T AddSimplifierAnnotations<T>( T node )
             where T : SyntaxNode
             => node.WithSimplifierAnnotation();
 
+        public AnonymousFunctionExpressionSyntax SimplifyAnonymousFunction<T>( T node )
+            where T : AnonymousFunctionExpressionSyntax
+            => node switch
+            {
+                SimpleLambdaExpressionSyntax { Block.Statements: [ExpressionStatementSyntax expressionStatement] } simpleLambdaExpression =>
+                    simpleLambdaExpression.Update(
+                        simpleLambdaExpression.AttributeLists,
+                        simpleLambdaExpression.Modifiers,
+                        simpleLambdaExpression.Parameter,
+                        SyntaxFactory.Token( SyntaxKind.EqualsGreaterThanToken ),
+                        null,
+                        expressionStatement.Expression ),
+                SimpleLambdaExpressionSyntax { Block.Statements: [ThrowStatementSyntax throwStatement] } simpleLambdaExpression =>
```

Comment:
`() => { throw; }` is invalid code, but it can still happen at design-time and we probably shouldn't throw an exception for it.
```suggestion
                SimpleLambdaExpressionSyntax { Block.Statements: [ThrowStatementSyntax { Expression: not null } throwStatement] } simpleLambdaExpression =>
```